### PR TITLE
ENIGMA-Kbase project update

### PR DESCRIPTION
**Pull request by psnovichkov**
This PR involves changes in widgets involved in ENIGMA-KBase integration project only.
Few changes are made in kbase-extension/static/narrative_paths.js

function_input/kbaseGrowthCurvesInput.js
function_input/kbaseGrowthParamsPlotInput.js
function_input/kbaseSampleProperty2DPlotInput.js
function_input/kbaseSamplePropertyHistogramInput.js
function_input/parameter_input/kbaseNarrativeParameterCustomTextSubdataInput.js

function_output/kbaseMatrix2DAbstract.js
function_output/kbaseChromatograms.js
function_output/kbaseChromatographyMatrix.js
function_output/kbaseGrowthCurves.js
function_output/kbaseGrowthMatrix.js
function_output/kbaseGrowthMatrixAbstract.js
function_output/kbaseGrowthParameters.js
function_output/kbaseGrowthParametersAbstract.js
function_output/kbaseGrowthParamsPlot.js
function_output/kbaseSampleProperty2DPlot.js
function_output/kbaseSamplePropertyHistogram.js
function_output/kbaseSamplePropertyMatrix.js
function_output/kbaseSamplePropertyMatrixAbstract.js

function_output/kbaseWellSample2DPlot.js
function_output/kbaseWellSampleHistogram.js

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/function_input/kbaseGrowthParamsPlotInput.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ define(['jquery',
         
         version: "1.0.0",
         options: {
-            loadingImage: "../images/ajax-loader.gif",
+            loadingImage: window.kbconfig.loading_gif,
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_input/kbaseGrowthParamsPlotInput.js]: Using window.kbconfig is deprecated, please start including narrativeConfig as a dependency, and using Config.get('loading_gif') for the spinner image. See here: https://github.com/kbase/narrative/blob/develop/docs/developer/widget_helper_modules.md#narrative-config for examples

File: kbase-extension/static/kbase/js/widgets/function_input/kbaseGrowthParamsPlotInput.js [hunk 2]
```diff
@@ -238,46 +236,61 @@ define(['jquery',
...
+                                        self.state = self.STATE_READY;
+                                        if(doneCallback) { doneCallback(self.conditionParams); }
+                                    },
+                                    function(error) {
+                                        console.error(error);
+                                    }
+                                );                        
                             },
-                            function(error) {
+                            function(eror){
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_input/kbaseGrowthParamsPlotInput.js]: This'll cause an error if an error gets thrown - the 'eror' name should be 'error' for the console message.

File: kbase-extension/static/kbase/js/widgets/function_input/kbaseSamplePropertyHistogramInput.js [hunk 3]
```diff
@@ -2,15 +2,19 @@
...
+    function( $, Config ) {
     $.KBWidget({
-        name: "kbaseWellSampleHistogramInput",
+        name: "kbaseSamplePropertyHistogramInput",
         parent: "kbaseNarrativeMethodInput",
         
         version: "1.0.0",
         options: {
-            loadingImage: "../images/ajax-loader.gif",
+            loadingImage: window.kbconfig.loading_gif,
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_input/kbaseSamplePropertyHistogramInput.js]: same here - Config.get('loading_gif')

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseMatrix2DAbstract.js [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+            this.$messagePane.hide();
+            this.$messagePane.empty();
+        },
+
+        clientError: function(error){
+            this.loading(false);
+            this.showMessage(error.error.error);
+        },            
+
+        uuid: function() {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseMatrix2DAbstract.js]: There's a utility module you can use for uuids (and some other things) in util/string. See https://github.com/kbase/narrative/blob/develop/docs/developer/widget_helper_modules.md#utilstring for an example.

**Comment by briehl** [comment]
Hey Pavel, I have a couple comments inline, but it looks pretty good.

File: kbase-extension/static/kbase/js/widgets/function_input/kbaseGrowthParamsPlotInput.js (hunk 2 above)
Comment (psnovichkov) [inline kbase-extension/static/kbase/js/widgets/function_input/kbaseGrowthParamsPlotInput.js]: that is why I hate javascript...

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseMatrix2DAbstract.js (hunk 4 above)
Comment (psnovichkov) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseMatrix2DAbstract.js]: I will do it later this week

**Comment by psnovichkov** [comment]
I switched to Config (both loadingImage and workspaceUrl), and fixed 'eror'.

**Comment by briehl** [comment]
The build is failing because qgrid can't be installed. This won't be a problem on CI, since that dependency is included in a subimage. Merging.

**Pull request merged**